float32 x
float32 y
float32 width
float32 height
float32 area
float32 angle